src/charm.py

```diff
@@ -166,12 +211,15 @@ def _on_certificates_relation_joined(self, event: RelationJoinedEvent) -> None:
             self.unit.status = WaitingStatus("Waiting for certificates relation to be created")
             event.defer()
             return
-        hostname = self.get_hostname()
-        if not hostname:
-            self.unit.status = BlockedStatus("Waiting for hostname to be configured")
-            event.defer()
+        try:
+            gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
+        except InvalidCharmConfigError as exc:
+            LOGGER.error("Invalid charm config: %s", exc.msg)
+            self.unit.status = BlockedStatus("Invalid charm configuration")
```

Comment:
Minor: maybe adding each configuration is invalid in the message or pointing to check the logs.

Reply:
It should already be in exc.msg but I overlooked this one, will update it. Thanks!